pkg/csi-common/nodeserver-default.go

```diff
@@ -17,16 +17,24 @@ limitations under the License.
 package csicommon
 
 import (
+	"fmt"
+	"os"
+
+	"github.com/ceph/ceph-csi/pkg/util"
+
 	"github.com/container-storage-interface/spec/lib/go/csi"
+	csipbv1 "github.com/container-storage-interface/spec/lib/go/csi"
 	"golang.org/x/net/context"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 	"k8s.io/klog"
+	"k8s.io/kubernetes/pkg/volume"
 )
 
 // DefaultNodeServer stores driver object
 type DefaultNodeServer struct {
 	Driver *CSIDriver
+	Type   string
```

Comment:
I did not follow why we need this passed to the NodeServer, is it used anywhere? Or, is it some form of future proofing?

Reply:
Its future proofing to treat different scenarios in this common func.